db/version_set.cc

```diff
@@ -1043,7 +1044,26 @@ void LevelIterator::Seek(const Slice& target) {
   if (file_iter_.iter() != nullptr) {
     file_iter_.Seek(target);
   }
-  SkipEmptyFileForward();
+  if (SkipEmptyFileForward() && prefix_extractor_ != nullptr &&
+      file_iter_.iter() != nullptr && file_iter_.Valid()) {
+    // We've skipped the file we initially positioned to. In the prefix
+    // seek case, it is likely that the file is skipped because of
+    // prefix bloom or hash, where more keys are skipped. We then check
+    // the current key and invalidate the iterator if the prefix is
```

Comment:
Based on the inline comment, it is not clear to me that if this is an optimization or bug fix.

Reply:
It's not an optimization nor bug. It's to make the internal interface clearer.

Reply:
What happens if we do not invalidate? Does it get invalidated anyway somewhere else? Is the goal to invalidate early?

The comment below says "This avoid LevelIterator to skip keys". So before the patch it did not skip the keys? Was it a bug?

Reply:
It may ever be invalidated, but it's still follow the contract. The goal is not to invalidate earlier. The purpose of the PR is not to put it in a very surprising location. Skipping keys is not a correctness bug if the keys skipped don't belong to the same prefix as the seek key.